staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/converter.go

```diff
@@ -396,8 +396,10 @@ func interfaceFromUnstructured(sv, dv reflect.Value) error {
 func (c *converterImpl) ToUnstructured(obj interface{}) (map[string]interface{}, error) {
 	var u map[string]interface{}
 	var err error
-	if unstr, ok := obj.(runtime.Unstructured); ok {
-		u = DeepCopyJSON(unstr.UnstructuredContent())
+	ro, ro_ok := obj.(runtime.Object)
+	if _, ok := obj.(runtime.Unstructured); ok && ro_ok {
```

Comment:
if u, ok := obj.(runtime.Unstructured); ok && ro_ok {

Reply:
Do we really need the first conversion?

Reply:
We need `.(runtime.Object)` because we want to call `DeepCopyObject()`. We need `.(runtime.Unstructured)` because we need `UnstructuredContent()`. Two distinct interfaces which we both need for this optimization so there have to be two checks.

Reply:
This confusing runtime.Unstructured/unstructured.Unstructured distinction. At that point you cannot import the later without a dep circle. Ugly, but necessary.

Reply:
Yes, there is a circle. I was thinking maybe we should merge two packages? But this seems unnecessary if `runtime.Unstructured` is available.

Reply:
let's leave it as it is. We can live with that ugliness for now.